azavea/climate-change-lab
# Feature/kak/threshold parameter controls#206

## Overview

Adds controls above chart for user modification of extra query parameters used by the three threshold indicators. See #206 and #182 for discussion.


### Demo

![image](https://user-images.githubusercontent.com/960264/29425166-8c51f164-8382-11e7-878c-90420de04a3d.png)


### Notes

Uses Angular [reactive forms](https://angular.io/guide/reactive-forms#reactive-forms), which does not allow for using a Bootstrap dropdown, as it generates its own Material Design styled `select` component. Updated styles to try to match, but the area around the drop-down caret looks slightly different.

The options for `threshold_units` and `theshold_comparator` are not programatically discoverable in the indicator description, as are other options, such as `units`. Currently the options and defaults are coded into the threshold parameters component. These could maybe be moved out to a shared constants file for indicators with extra parameters in implementing the next set.

There seem to be some strange behavior around removing charts and re-adding them that may warrant opening a new issue.

Created #209 for saving the new control values to the project.

## Testing Instructions

 * Select an indicator from the sidebar whose name ends with 'Threshold' (there are three)
 * Should get controls above chart
 * Changing the controls should update the chart as expected

Closes #206.
